core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -892,6 +1044,299 @@ export class LoggedInUser extends BaseUser {
+    await this.clickButtonToNavigateToNewPage(
+      'Browse our Lessons',
+      'Creator Guidelines in the About Menu on navbar',
+      mathClassroomUrl,
+      'Creator Guidelines'
+    );
+  }
+
+  /**
+   * Clicks the link with the text "Access the Android app" on the Teach page.
+   */
+  async clickAccessTheAndroidAppOnTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      'Access the Android app',
+      'Creator Guidelines in the About Menu on navbar',
+      androidUrl,
+      'Creator Guidelines'
+    );
+  }
+
+  /**
+   * Clicks the link with the text "VISIT CLASSROOM" on the Teach page.
+   */
+  async clickVisitClassroomOnTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      'VISIT CLASSROOM',
+      'Creator Guidelines in the About Menu on navbar',
+      mathClassroomUrl,
+      'Creator Guidelines'
+    );
+  }
+
+  /**
+   * Clicks the link with the text "BROWSE LIBRARY" on the Teach page.
+   */
+  async clickBrowseLibraryOnTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      'BROWSE LIBRARY',
+      'Creator Guidelines in the About Menu on navbar',
+      communityLibraryUrl,
+      'Creator Guidelines'
+    );
+  }
+
+  /**
+   * Clicks the link with the text "Explore Lessons" on the Teach page.
+   */
+  async clickExploreLessonsOnTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      'Explore lessons',
+      'Creator Guidelines in the About Menu on navbar',
+      mathClassroomUrl,
+      'Creator Guidelines'
+    );
+  }
+
+  /**
+   * Clicks the link on the Terms page about the CC-BY-SA 4.0 license.
+   */
+  async clickLinkToLicenseOnTermsPage(): Promise<void> {
+    await this.page.waitForXPath('(//a[contains(text(),"here")])[1]');
+
+    const [link] = await this.page.$x('(//a[contains(text(),"here")])[1]');
+
+    await Promise.all([this.page.waitForNavigation(), await link.click()]);
+
+    expect(this.page.url()).toBe(
+      'https://creativecommons.org/licenses/by-sa/4.0/legalcode'
+    );
+  }
+
+  /**
+   * Clicks the link on the Terms page that leads to the Oppia Announce google group.
+   */
+  async clickLinkToGoogleGroupOnTermsPage(): Promise<void> {
+    await this.page.waitForXPath('(//a[contains(text(),"here")])[2]');
+
+    const [link] = await this.page.$x('(//a[contains(text(),"here")])[2]');
+
+    await Promise.all([this.page.waitForNavigation(), await link.click()]);
+
+    expect(this.page.url()).toBe(googleGroupsOppiaAnnouceUrl);
+  }
+
+  /**
+   * Clicks the link on the Privacy Policy page that goes to the home page.
+   */
+  async clickLinkToHomePageOnPrivacyPolicyPage(): Promise<void> {
+    await this.page.waitForXPath(
+      '//a[contains(text(),"https://www.oppia.org")]'
+    );
+    await Promise.all([
+      this.page.waitForNavigation(),
+      this.clickOn('https://www.oppia.org'),
+    ]);
+
+    //  Wait for redirect.
+    await this.page.waitForNetworkIdle();
```

Comment:
Yes. Otherwise, it will be flakey as it will sometimes check the URL before the redirect. The waitForNavigation is resolved once the homepage loads.

Though I've changed the waitForNavigation call to be

```
this.page.waitForNavigation({waitUntil: 'networkidle0'}),
```

Which should have the same result.